da/celestia/proto/celestia.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+package celestia;
+
+option go_package = "github.com/celestiaorg/optimint/da/celestia/pb";
+import "tendermint/types/types.proto";
+
+service DataRetriever {
+	rpc RetrieveShares (Request) returns (Shares) {}
+	rpc RetrieveStateRelevantMessages (Request) returns (StateRevelantMessages) {}
+	rpc RetrieveMessages (Request) returns (tendermint.types.Messages) {}
+}
+
+message DataAvailabilityHeader {
+	repeated bytes row_roots = 1;
+	repeated bytes column_roots = 2;
+}
+
+message Request {
+	// NamespaceID identifies the namespace in Celestia chain.
+	bytes namespace_id = 1;
+
+	// DataAvailabilityHeader identifies the block in Celestia chain.
+	tendermint.types.DataAvailabilityHeader dah = 2;
```

Comment:
Such a `Request` does not need the full `DataAvailabilityHeader`. 

Instead, the block data should be either identified and requested by the data root ([which is part of the header](https://github.com/celestiaorg/celestia-core/blob/c2df2c4ce0942b8e974fd2b530355f2a58139926/types/block.go#L355)) and is sufficient to uniquely identify the data (it is the [Hash of the dah](https://github.com/celestiaorg/celestia-core/blob/c2df2c4ce0942b8e974fd2b530355f2a58139926/types/block.go#L1040-L1063)).

Alternatively, shares of a namespace could also be requested by height.